src/BSON/UTCDateTime.c

```diff
@@ -47,17 +52,58 @@ PHONGO_API zend_class_entry *php_phongo_utcdatetime_ce;
 
 zend_object_handlers php_phongo_handler_utcdatetime;
 
-/* {{{ proto BSON\UTCDateTime UTCDateTime::__construct(integer $milliseconds)
+static zend_class_entry *php_phongo_utcdatetime_get_date_ce(TSRMLS_D)
+{
+	zend_class_entry *found_ce = NULL;
+
+	#if PHP_VERSION_ID >= 70000
+		zend_string *zs_classname = zend_string_init("DateTimeInterface", sizeof("DateTimeInterface")-1, 0);
+		found_ce = zend_fetch_class(zs_classname, ZEND_FETCH_CLASS_AUTO|ZEND_FETCH_CLASS_NO_AUTOLOAD|ZEND_FETCH_CLASS_SILENT);
+		zend_string_free(zs_classname);
+	#else
+		found_ce = zend_fetch_class("DateTimeInterface", sizeof("DateTimeInterface")-1, ZEND_FETCH_CLASS_AUTO|ZEND_FETCH_CLASS_NO_AUTOLOAD|ZEND_FETCH_CLASS_SILENT TSRMLS_CC);
+	#endif
+
+	return found_ce ? found_ce : php_date_get_date_ce();
```

Comment:
@derickr: Would it make sense to memoize this function by saving `found_ce` in a static global? I also thought about initializing the constructors `date_ce` in UTCDateTime's MINIT, but that assumes we are always initialized after PHP's date extension. I'd like to respect the edge case where that may not be true.